Test/boogie/variableNames.dfy

```diff
@@ -0,0 +1,38 @@
+// RUN: %dafny /compile:0 /env:0 /print:- "%s" > "%t"
```

Comment:
Hmm, on the one hand I'm glad you've added a thorough test case for this, but on the other hand I'm worried this test is going to be very brittle and constantly breaking with every translation change. We don't generally assert the exact output of any compilers or the translator, just the properties that we actually care about (verification or runtime behavior).

Would it be possible to turn this into a unit test for something like `Translator.TrStmtList`, so we're at least not asserting the exact content of `DafnyPrelude.bpl`?

Reply:
> We don't generally assert the exact output of any compilers or the translator, just the properties that we actually care about (verification or runtime behavior).

Yes, but the feature this PR implements is different in nature than existing features. It's a feature that doesn't influence the verification or runtime behavior. It's even hard to specify what this change does. The intent is to more or less reduce the diff of two Boogie programs when a Dafny program is changed.

I can try to put that into a test, and then expect the result of the diff. I think such a test would be quite resistant to changes in our Boogie output since then both sides of the diff would change.